week1/assignment1_4_1/exercise1/main.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+Exercise 1
+
+Write a C++ program that capitalizes the first letter of the string and sets the rest as lowercase letters.
+Use the following main function and write a function that passes the assertions.
+*/
+
+#include <cassert>          //for assert
+#include <cctype>           //for toupper and tolower
+#include <iostream>
+#include <string>
+#include <string_view>
+
+//macro for disabling asserts
+//#define NDEBUG
+
+//Function capitalize_string capitalizes the first character of every word. Return a sting. 
+//Takes input as string_view, as it is only copied once so this is an effective way to do it. 
+std::string capitalize_string(std::string_view unmodified)
+{
+    std::string capitalized{ unmodified };
+
+    //Loop, and check with a bool, if should be capitalized or un-capitalized
+    //A space tells that the next should be capitalized
+    for (bool capitalize_next{ true }; auto & c : capitalized)
+    {
+        if (capitalize_next)
+        {
+            c = std::toupper(static_cast<unsigned char>(c));
+            capitalize_next = false;
+        }
+        else if (c == ' ')
+        {
+            capitalize_next = true;
+        }
+        else
+        {
+            c = std::tolower(static_cast<unsigned char>(c));
+        }
```

Comment:
You've nicely figured out how to change the first letter of each word to capital letters, even though this didn't quite match the assignment (?), or maybe I misunderstood myself :D